module/synchronization/core.go

```diff
@@ -241,8 +241,9 @@ func (c *Core) prune(final *flow.Header) {
 		}
 	}
 
-	prunedHeights := len(c.heights) - initialHeights
-	prunedBlockIDs := len(c.blockIDs) - initialBlockIDs
+	prunedHeights := initialHeights - len(c.heights)
```

Comment:
I saw the log printing the prunedHeights was negative, and realized the math here needs to reversed.